### PR TITLE
Rename requirements to dependencies

### PR DESCRIPTION
**Pull request by richtja**
Based on the BP004 this change will expand the usability of this term
because, The “dependency” word has a high-level meaning and is less
misinterpreted when compared to “requirement”. The idea is not to use a
strict math definition of a dependency relation, but a soft meaning
where after one task runs, another should run, and the second cannot
start if the first has not finished.

Reference: #5169
Signed-off-by: Jan Richter <jarichte@redhat.com>

### CODE REVIEW

File: avocado/core/runners/requirement_asset.py [hunk 1]
```diff
@@ -101,8 +101,8 @@ def run(self):
 
 class RunnerApp(nrunner.BaseRunnerApp):
     PROG_NAME = 'avocado-runner-requirement-asset'
```
Comment (ana) [inline avocado/core/runners/requirement_asset.py:103]: shouldn't be this one also renamed to `avocado-runner-package`?

File: avocado/core/task/runtime.py [hunk 2]
```diff
@@ -114,35 +114,42 @@ def get_requirements_form_runnable(cls, runnable, status_server_uri=None,
...
+                                   job_id=job_id)
+            dependencies_runtime_tasks.append(cls(dependence_task))
 
-        return requirements_runtime_tasks
+        return dependencies_runtime_tasks
+
+    def is_dependencies_finished(self):
+        for dependencie in self.dependencies:
+            if not dependencie.status or not ("FINISHED" in dependencie.status
+                                              or "FAILED" in dependencie.status):
```
Comment (ana) [inline avocado/core/task/runtime.py:150]: if you don't mind `dependencie` -> `dependency`

File: avocado/core/task/runtime.py [hunk 3]
```diff
@@ -181,21 +188,20 @@ def __init__(self, tests, test_suite_name, status_server_uri, job_id):
...
 
-    def _connect_requirements_with_test(self, requirements, runtime_test):
-        for requirement_task in requirements:
-            if requirement_task in self.graph:
-                requirement_task = self.graph.get(requirement_task)
+    def _connect_dependencies_with_test(self, dependencies, runtime_test):
+        for dependence_task in dependencies:
+            if dependence_task in self.graph:
+                dependence_task = self.graph.get(dependence_task)
             else:
```
Comment (ana) [inline avocado/core/task/runtime.py:202]: if you don't mind `dependence` -> `dependency`

File: avocado/core/task/runtime.py (hunk 2 above)
Comment (ana) [reply avocado/core/task/runtime.py:150]: In second thought, I think it's [dependence](https://en.wiktionary.org/wiki/dependence#English)) and the plural would be `dependences`.

**Comment by ana** [comment]
* There are some leftovers in `selftests/unit/test_runner_requirement_package.py` and `selftests/unit/test_runner_requirement_asset.py`. Grep for `requirement-package` and `requirement-asset` and you'll see them.

* As you know I'm not a native English speaker. Your changes are mixing dependency and dependence, with their plural versions. They don't mean the same and from their definitions https://en.wiktionary.org/wiki/dependency and https://en.wiktionary.org/wiki/dependence I think we should stick to the latter word.

File: avocado/core/task/runtime.py (hunk 2 above)
Comment (richtja) [reply avocado/core/task/runtime.py:150]: Actually I am not really sure what is the exact difference between `dependency` and `dependence`. But it looks like in our context the `dependency` is [better](https://ell.stackexchange.com/questions/41528/differences-between-dependence-and-dependency).

**Comment by richtja** [comment]
> * There are some leftovers in `selftests/unit/test_runner_requirement_package.py` and `selftests/unit/test_runner_requirement_asset.py`. Grep for `requirement-package` and `requirement-asset` and you'll see them.
> 
>     * As you know I'm not a native English speaker. Your changes are mixing dependency and dependence, with their plural versions. They don't mean the same and from their definitions https://en.wiktionary.org/wiki/dependency and https://en.wiktionary.org/wiki/dependence I think we should stick to the latter word. Or at least, use consistently always the same.

Hi @ana thanks, for your review. I missed those `requirement-package` and `requirement-asset` I will fix them.

**Review by clebergnu — CHANGES_REQUESTED** [review]
Hi @richtja ,

I agree with the name change, and I like the fact that this makes the "requirement-*" runners less special.

But, I still see some references to "requirements", such as:

```
avocado/core/plugin_interfaces.py:    async def check_task_requirements(runtime_task):
avocado/core/task/statemachine.py:        # a task waiting requirements already checked its requirements
```

Any reason for not changing those?

**Comment by richtja** [comment]
> Hi @richtja ,
> 
> I agree with the name change, and I like the fact that this makes the "requirement-*" runners less special.
> 
> But, I still see some references to "requirements", such as:
> 
> ```
> avocado/core/plugin_interfaces.py:    async def check_task_requirements(runtime_task):
> avocado/core/task/statemachine.py:        # a task waiting requirements already checked its requirements
> ```
> 
> Any reason for not changing those?

Hi @clebergnu, yes, I left those as requirements intentionally. IMO, the method `check_task_requirements` is not checking the Task's dependencies (Tasks which have to be run before the specific Task), but it is checking if it is possible to run the Task in spawner environment (e.g. if specific avocado runner exists). Because of this,  those “requirements” are different from Task dependencies. I didn't find a better name, so I left it as `requirements`, but if you have some better naming, I would appreciate it.

**Comment by clebergnu** [comment]
> > Hi @richtja ,
> > I agree with the name change, and I like the fact that this makes the "requirement-*" runners less special.
> > But, I still see some references to "requirements", such as:
> > ```
> > avocado/core/plugin_interfaces.py:    async def check_task_requirements(runtime_task):
> > avocado/core/task/statemachine.py:        # a task waiting requirements already checked its requirements
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Any reason for not changing those?
> 
> Hi @clebergnu, yes, I left those as requirements intentionally. IMO, the method `check_task_requirements` is not checking the Task's dependencies (Tasks which have to be run before the specific Task), but it is checking if it is possible to run the Task in spawner environment (e.g. if specific avocado runner exists). Because of this, those “requirements” are different from Task dependencies. I didn't find a better name, so I left it as `requirements`, but if you have some better naming, I would appreciate it.

You are right, good catch!

**Review by clebergnu — APPROVED** [review]
LGTM, thanks!

**Pull request merged**